### ISSUE TITLE
Missing license id in bom.xml

### ISSUE THREAD

**Issue by dalikrisz** [CycloneDX/cyclonedx-maven-plugin#132]
I am using cyclonedx-maven-plugin 2.5.3 to generate my bom.xml.
In the generated file there are some missing license information. 

Let me show you an example. I have this in my pom.xml:
```
<dependency>
   <groupId>com.itextpdf</groupId>
   <artifactId>itextpdf</artifactId>
   <version>5.5.13.2</version>
</dependency>
```

In my understanding this should produce a bom.xml where the licenses tag contains the following:
```
<license>
    <id>AGPL-3.0</id>
    <name>GNU Affero General Public License v3</name>
    <url>http://www.fsf.org/licensing/licenses/agpl-3.0.html</url>
</license>
```

Instead, I get the following (note the missing license id):
```
<license>
    <name>GNU Affero General Public License v3</name>
    <url>http://www.fsf.org/licensing/licenses/agpl-3.0.html</url>
</license>
```

This is an issue for me, as Dependency Track will only recognize the license if it has an id. This way I cannot use policies to alert me when the project uses an AGPL licence.

Maybe the problem is the missing [license-mapping](https://github.com/CycloneDX/cyclonedx-core-java/blob/master/src/main/resources/license-mapping.json) for this license?

Can anyone please help me with this issue?

**Comment by stevespringett** [comment]
> In my understanding this should produce a bom.xml where the licenses tag contains the following:
>```
> <license>
>    <id>AGPL-3.0</id>
>    <name>GNU Affero General Public License v3</name>
>    <url>http://www.fsf.org/licensing/licenses/agpl-3.0.html</url>
> </license>```

That is incorrect. A license can only contain an id or a name, but not both.

You're getting a license name of `GNU Affero General Public License v3` because that name is not an exact match to any known SPDX license names. Possible SPDX license names and their identifiers are:

| SPDX License Name  |  SPDX License ID  |
|-------------------------------------------------|-------------------|
| GNU Affero General Public License v3.0 only  | AGPL-3.0-only |
| GNU Affero General Public License v3.0 or later | AGPL-3.0-or-later |
| GNU Affero General Public License v3.0   | AGPL-3.0  |

Refer to https://spdx.org/licenses/

Had it been an exact match, the BOM would have an id rather than a name. In some cases, CycloneDX can make make a confident determination on which SPDX license to resolve. In a case like this where there's three possibilities, CycloneDX does not make any assumptions.

**Comment by dalikrisz** [comment]
Thank you for your fast answer.

If I understand correctly, this license name is baked into the dependency artifact (in this case the itextpdf), and CycloneDX tries to match this to a SPDX license.

So if my bom.xml contains more license names, that means that those license names are not valid SPDX names, and there is nothing I can do to "fix" these. The only solution is to change the name to id either by hand or a script, after the bom generation, right?

**Comment by stevespringett** [comment]
That's correct. The Maven plugin is built on top of [CycloneDX Core Java](https://github.com/CycloneDX/cyclonedx-core-java) which is a library that serializes/deserializes the model and provide validation capabilities. The Core Java library also contains a mapping file which contains license names commonly used in a pom, and the mapping to the SPDX license ID or expression. So it's possible to improve the mapping as well.

https://github.com/CycloneDX/cyclonedx-core-java/blob/master/src/main/resources/license-mapping.json
